core/src/main/java/org/geysermc/geyser/translator/protocol/bedrock/BedrockCommandRequestTranslator.java

```diff
@@ -39,15 +40,36 @@ public class BedrockCommandRequestTranslator extends PacketTranslator<CommandReq
     @Override
     public void translate(GeyserSession session, CommandRequestPacket packet) {
         String command = MessageTranslator.convertToPlainText(packet.getCommand());
-        if (!(session.getGeyser().getPlatformType() == PlatformType.STANDALONE
-                && GeyserImpl.getInstance().commandManager().runCommand(session, command.substring(1)))) {
-            if (MessageTranslator.isTooLong(command, session)) {
-                return;
+
+        // remove the beginning slash
+        command = command.substring(1);
+
+        // running commands via Bedrock's command select menu adds a trailing whitespace which Java doesn't like
+        // https://github.com/GeyserMC/Geyser/issues/3877
+        command = command.stripTrailing();
+
+        if (session.getGeyser().getPlatformType() == PlatformType.STANDALONE) {
+            // try to handle the command within the standalone command manager
+
+            String[] args = command.split(" ");
+            if (args.length > 0) {
+                String root = args[0];
+
+                // todo cloud do we want to pass the command to the server
+                //  if cloud gives a NoSuchCommandException? might be more accurate.
+                CommandRegistry registry = GeyserImpl.getInstance().commandRegistry();
+                if (registry.cloud().rootCommands().contains(root)) {
+                    // todo cloud might not like the trailing whitespace either
+                    registry.runCommand(session, command);
+                    return; // don't pass the command to the java server
```

Comment:
```suggestion
        // Remove the beginning slash
        command = command.substring(1);

        // Running commands via Bedrock's command select menu adds a trailing whitespace which Java doesn't like
        // https://github.com/GeyserMC/Geyser/issues/3877
        command = command.stripTrailing();

        if (session.getGeyser().getPlatformType() == PlatformType.STANDALONE) {
            // Try to handle the command within the standalone command manager

            String[] args = command.split(" ");
            if (args.length > 0) {
                String root = args[0];

                // TODO cloud do we want to pass the command to the server
                //  if cloud gives a NoSuchCommandException? might be more accurate.
                CommandRegistry registry = GeyserImpl.getInstance().commandRegistry();
                if (registry.cloud().rootCommands().contains(root)) {
                    // TODO cloud might not like the trailing whitespace either
                    registry.runCommand(session, command);
                    return; // Don't pass the command to the java server
```

Reply:
The first TODO marker here I'm not sure what it's suggesting, is this done or can it be reworded?

Reply:
it would be theoretically more flexible than just checking the root command but i'm not sure if its worth it. and it would require making the exception handling code more dynamic (in a way that isn't intrinsically better)

i'll remove it unless we run into it issues - the current code only checks the root as well